app/common/util/src/main/java/io/syndesis/common/util/Names.java

```diff
@@ -16,36 +16,40 @@
 package io.syndesis.common.util;
 
 import java.util.Locale;
+import static io.syndesis.common.util.Strings.truncate;
 
 public final class Names {
 
-    private static final String INVALID_CHARACTER_REGEX = "[^a-zA-Z0-9-]";
+    private static final String VALID_REGEX = "[-A-Za-z0-9\\.]+";
+    private static final String INVALID_CHARACTER_REGEX = "[^a-zA-Z0-9-\\.]";
     private static final String SPACE = " ";
     private static final String BLANK = "";
     private static final String DASH = "-";
-    private static final int MAXIMUM_NAME_LENGTH = 100;
+    //The actual limit is 253,
+    // but individual resources have other limitations
+    // (e.g. service has 63: RFC 1035)
+    private static final int MAXIMUM_NAME_LENGTH = 63;
 
     private Names() {
         //Utility
     }
 
     /**
      * Sanitizes the specified name by applying the following rules:
-     * 1. Keep the first 100 characters.
-     * 2. Replace spaces with dashes.
-     * 3. Remove invalid characters.
+     * 1. Replace spaces with dashes.
+     * 2. Remove invalid characters.
+     * 3. Keep the first 64 characters.
      * @param name  The specified name.
      * @return      The sanitized string.
      */
     public static String sanitize(String name) {
-        String trimmed = name.length() > MAXIMUM_NAME_LENGTH ? name.substring(0, MAXIMUM_NAME_LENGTH) : name;
-
-        return trimmed
+        return truncate(name
             .replaceAll(SPACE, DASH)
             .replaceAll(INVALID_CHARACTER_REGEX, BLANK)
             .toLowerCase(Locale.US)
             .chars()
             .filter(i -> !String.valueOf(i).matches(INVALID_CHARACTER_REGEX))
```

Comment:
same comment as for `Labels`: Suggest character matching instead of regexp for single chars (remember, its expensive to compile a pattern every time a fresh).